lib/msal-browser/src/request/RequestHelpers.ts

```diff
@@ -98,3 +101,34 @@ export async function initializeSilentRequest(
         forceRefresh: request.forceRefresh || false,
     };
 }
+
+/**
+ * Validates that the combination of request method, protocol mode and authorize body parameters is correct.
+ * Returns the validated or defaulted HTTP method or throws if the configured combination is invalid.
+ * @param interactionRequest
+ * @param protocolMode
+ * @returns
+ */
+export function validateRequestMethod(
+    interactionRequest: BaseAuthRequest | PopupRequest | RedirectRequest,
+    protocolMode: ProtocolMode
+): Constants.HttpMethod {
+    let httpMethod: Constants.HttpMethod | undefined;
+    const requestMethod = interactionRequest.httpMethod;
+
+    if (protocolMode === ProtocolMode.EAR) {
+        // Don't override httpMethod if it is already set, default to POST if not set
+        httpMethod = requestMethod || Constants.HttpMethod.POST;
+        // Validate that method is not GET if protocol mode is EAR
+        if (httpMethod !== Constants.HttpMethod.POST) {
+            throw createClientConfigurationError(
+                ClientConfigurationErrorCodes.invalidRequestMethodForEAR
+            );
+        }
```

Comment:
nit: Suggest avoiding the `requestMethod || Constants.HttpMethod.POST` if we only allow POST anyway. I think this makes it a little clearer what the expectations are

```suggestion
    if (protocolMode === ProtocolMode.EAR) {
        // Validate that method can only be POST when protocol mode is EAR
        if (requestMethod && requestMethod !== Constants.HttpMethod.POST) {
            throw createClientConfigurationError(
                ClientConfigurationErrorCodes.invalidRequestMethodForEAR
            );
        } else {
          httpMethod = Constants.HttpMethod.POST;
        }
```